api_client.go

```diff
@@ -0,0 +1,1031 @@
+)
+
+// ErrorResponse defines the error that will be asserted by FGA API.
+// This will only be used for error that is not defined
+type ErrorResponse struct {
+    Code    string `json:"code"`
+    Message string `json:"message"`
+}
+
+// APIClient manages communication with the OpenFGA API v0.1
+// In most cases there should be only one, shared, APIClient.
+type APIClient struct {
+    cfg    *Configuration
+    common service // Reuse a single struct instead of allocating one for each service on the heap.
+
+    // API Services
+
+    OpenFgaApi OpenFgaApi
+}
+
+type service struct {
+    client      *APIClient
+    RetryParams *RetryParams
+}
+
+// NewAPIClient creates a new API client. Requires a userAgent string describing your application.
+// optionally a custom http.Client to allow for advanced features such as caching.
+func NewAPIClient(cfg *Configuration) *APIClient {
+
+    if cfg.HTTPClient == nil {
+        if cfg.Credentials == nil {
+            cfg.HTTPClient = http.DefaultClient
+        } else {
+            var httpClient, headers = cfg.Credentials.GetHttpClientAndHeaderOverrides()
+            if len(headers) > 0 {
+                for idx := range headers {
+                    cfg.AddDefaultHeader(headers[idx].Key, headers[idx].Value)
+                }
+            }
+            if httpClient != nil {
+                cfg.HTTPClient = httpClient
+            }
+        }
+    }
+
+    c := &APIClient{}
+    c.cfg = cfg
+    c.common.client = c
+    c.common.RetryParams = cfg.RetryParams
+
+    // API Services
+    c.OpenFgaApi = (*OpenFgaApiService)(&c.common)
+
+    if cfg.HTTPClient == nil {
+        if cfg.Credentials == nil {
+            cfg.HTTPClient = http.DefaultClient
+        } else {
+            var httpClient, headers = cfg.Credentials.GetHttpClientAndHeaderOverrides()
+            if len(headers) > 0 {
+                for idx := range headers {
+                    cfg.AddDefaultHeader(headers[idx].Key, headers[idx].Value)
+                }
+            }
+            if httpClient != nil {
+                cfg.HTTPClient = httpClient
+            }
+        }
+    }
+
+    return c
+}
+
+func (a APIClient) GetStoreId() string {
+    return a.cfg.StoreId
+}
+
+func (a APIClient) SetStoreId(storeId string) {
+    a.cfg.StoreId = storeId
+}
+
+func atoi(in string) (int, error) {
+    return strconv.Atoi(in)
+}
+
+// selectHeaderContentType select a content type from the available list.
+func selectHeaderContentType(contentTypes []string) string {
+    if len(contentTypes) == 0 {
+        return ""
+    }
+    if contains(contentTypes, "application/json") {
+        return "application/json"
+    }
+    return contentTypes[0] // use the first content type specified in 'consumes'
+}
+
+// selectHeaderAccept join all accept types and return
+func selectHeaderAccept(accepts []string) string {
+    if len(accepts) == 0 {
+        return ""
+    }
+
+    if contains(accepts, "application/json") {
```

Comment:
just to confirm, if the accepts contains application/json, we will not return the joined values (i.e., only return application/json alone)?